providers/common/include/internal/providercommonerr.h

```diff
@@ -51,24 +51,49 @@ int ERR_load_PROV_strings(void);
  */
 # define PROV_R_AES_KEY_SETUP_FAILED                      101
 # define PROV_R_BAD_DECRYPT                               100
+# define PROV_R_BAD_ENCODING                              141
+# define PROV_R_BAD_LENGTH                                142
+# define PROV_R_BOTH_MODE_AND_MODE_INT                    121
 # define PROV_R_CIPHER_OPERATION_FAILED                   102
 # define PROV_R_FAILED_TO_GET_PARAMETER                   103
 # define PROV_R_FAILED_TO_SET_PARAMETER                   104
+# define PROV_R_INAVLID_UKM_LEN                           143
 # define PROV_R_INVALID_AAD                               108
 # define PROV_R_INVALID_CUSTOM_LENGTH                     111
 # define PROV_R_INVALID_DATA                              115
+# define PROV_R_INVALID_DIGEST                            122
+# define PROV_R_INVALID_ITERATION_COUNT                   123
 # define PROV_R_INVALID_IVLEN                             116
 # define PROV_R_INVALID_IV_LENGTH                         109
 # define PROV_R_INVALID_KEYLEN                            117
+# define PROV_R_INVALID_KEY_LEN                           124
```

Comment:
At some point we need to fix this up.. (maybe not in this PR)

Reply:
Yeah.  There are some in the MAC that are both PROV_R_ codes and ERR_R_ codes...

Plus the IV lengths.

Reply:
I blame you 😁 

Ciphers are responsible KEYLEN and ciphers, KDFs and MACs for the KEY_LEN.
Ciphers are responsible for both IVLEN and IV_LENGTH.

I suspect they should be changed to KEY_LENGTH and IV_LENGTH respectively.

Reply:
What the hell?  Didn't I fix that up already?  I distinctly remember doing that as part of some PR, but maybe that hasn't been merged yet...

Reply:
Anyway, I agree with `KEY_LENGTH` and `IV_LENGTH`.  There's no reason to abbreviate error reason messages.

Reply:
Oh, please treat the `UKM_LEN` and `SALT_LEN` reasons similarly

Reply:
UKM_LEN fixed, SALT_LEN not done because it isn't KDF related.

Reply:
I.e. different PR for SALT_LEN

Reply:
Scratch that, it is KDF related and has been fixed now.

Reply:
However, INVALID_KEYLEN is a cipher issue.  So different PR.

Reply:
Issue  #9766 created so this doesn't get lost.